sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/package.scala

```diff
@@ -202,6 +202,26 @@ package object util extends Logging {
   /** Shorthand for calling truncatedString() without start or end strings. */
   def truncatedString[T](seq: Seq[T], sep: String): String = truncatedString(seq, "", sep, "")
 
+  /** Whether we have warned about plan string truncation yet. */
+  private val planSizeWarningPrinted = new AtomicBoolean(false)
+
+  def withSizeLimitedWriter[T](writer: Writer)(f: (Writer) => T): Option[T] = {
+    try {
+      val limited = new SizeLimitedWriter(writer, SQLConf.get.maxPlanStringLength)
+      Some(f(limited))
+    }
+    catch {
+      case e: WriterSizeException =>
+        writer.write("...")
```

Comment:
nit: Would we want to restrict string's length to maxPlanStringLength including `...`? I think exceeding defined length by max 3 chars is not a big deal but just to double check.

Reply:
It was easy enough to do this by dropping the limit by 3 characters.